src/kademlia/dht_tracker.cpp

```diff
@@ -155,9 +163,10 @@ namespace libtorrent { namespace dht
 		m_refresh_timer.expires_from_now(seconds(5), ec);
 		m_refresh_timer.async_wait(std::bind(&dht_tracker::refresh_timeout, self(), _1));
 
-		m_dht.bootstrap(m_state.nodes, f);
+		// bootstrap with mix of IP protocols via want/nodes/nodes6
+		m_dht.bootstrap(concat(m_state.nodes, m_state.nodes6), f);
```

Comment:
I don't think sending is FIFO. as far as I can tell, the traversal algorithm inserts nodes sorted. There's a ``std::lower_bound()`` in ``traversal_algorithm::add_entry()``.

In fact, looking at the bootstrap algorithm, it doesn't look quite right. It will just generate random node IDs for the IPs we bootstrap from, so it will be shuffled. But then we call ``trim_seed_nodes()``, which assumes that the nodes are sorted and have correct node IDs. The rationale being that we want to start farther away when bootstrapping.

Starting farther away makes sense, but I don't think this call makes sense, since we know that we just created random node IDs, and the first 32 nodes are just some random nodes, not the ones closest to us.